tsinfer/inference.py

```diff
@@ -821,25 +827,87 @@ def get_samples_tree_sequence(self):
         # Add in the remaining non-sample nodes.
         for u in range(self.sample_ids[-1] + 1, tsb.num_nodes):
             tables.nodes.add_row(flags=flags[u], time=time[u])
-
+        logger.debug("SAMPLE IDS: {}".format(self.sample_ids))
         logger.debug("Adding tree sequence edges")
         tables.edges.clear()
         left, right, parent, child = tsb.dump_edges()
-        if np.all(inference_sites == 0):
-            # We have no inference sites, so no edges have been estimated. To ensure
-            # we have a rooted tree, we add in edges for each sample to an artificial
-            # root.
-            assert left.shape[0] == 0
+        logger.debug("Checking for the left/right extent of all available data")
+        # Store the leftmost and rightmost extent of edges leading to TS nodes
```

Comment:
There is some slightly hairy logic starting from this line until line 900, which deals with edges whose children are samples that have missing data on the left and right. I would appreciate talking this through with @jeromekelleher when he returns to Oxford.